src/presentation/http/router/noteSettings.ts

```diff
@@ -89,6 +90,34 @@ const NoteSettingsRouter: FastifyPluginCallback<NoteSettingsRouterOptions> = (fa
     return reply.send(noteSettings);
   });
 
+  /**
+   * patch team member role by user and note id
+   */
+  fastify.patch<{
+    Body: {
+      userId: User['id'];
+      noteId: NoteInternalId,
+    }
+    Params: {
+      newRole: keyof typeof MemberRole,
```

Comment:
Why are you use `keyof typeof`?

Reply:
We use it to have list of keys of `MemberRole`, so newRole and Reply would be "read" | "write"